### PR TITLE
Add Python, C++, and Java Solution to Who's Opposite? (CF)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
all test cases pass for this solution :) 

@SansPapyrus683 hopefully my c++ code is better this time 🙏

### CODE REVIEW

**Comment by nathangong** [comment]
I can write up a java solution for this if u want

**Comment by ryanchou-dev** [comment]
> I can write up a java solution for this if u want

sure! i think you can just directly commit. it's just math, so it probably won't take that long to implement :)

File: solutions/cf-1560B.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+**Time Complexity:** $\mathcal{O}(1)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <stdlib.h>
```
Comment (SansPapyrus683) [inline solutions/cf-1560B.mdx:30]: what's this for now

File: solutions/cf-1560B.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+int solve(long a, long b, long c) {
+	long numGroup = abs(a - b) * 2;
+	
+	if (a > numGroup || b > numGroup || c > numGroup) {
+		return -1;
+	} else {
+		long ans = c + abs(a - b);
+		/*
+		* If the answer is greater than the number
```
Comment (SansPapyrus683) [inline solutions/cf-1560B.mdx:41]: add a space before the star to make them aligned with the top one

File: solutions/cf-1560B.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		subtract the number of people between a and b
+		to find the answer (since this creates a circle).
+		"""
+		if ans > num_group: ans -= (abs(a - b) * 2)
+		return ans
+
+n = int(input())
+for _ in range(n):
+	a, b, c = map(int, input().split())
+	print(solve(a, b, c))
```
Comment (SansPapyrus683) [inline solutions/cf-1560B.mdx:83]:
```suggestion
	print(solve(*map(int, input().split())))
```
just a wild idea lmao

File: solutions/cf-1560B.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		if (ans > numGroup) ans -= (abs(a - b) * 2);
+		return ans;
+	}
+
+}
+
+int main() {
+	int n; cin >> n;
+	for (int i = 0; i < n; i++) {
+		long a, b, c; cin >> a >> b >> c;
```
Comment (SansPapyrus683) [inline solutions/cf-1560B.mdx:55]: statements on one line
also if you don't want overflow, use `long long`- `long` is still 32 bits

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
i've seen worse

File: solutions/cf-1560B.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:30]: iostream is for input/output and
stdlib.h is for abs and stuff

File: solutions/cf-1560B.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:128]: errr i think listing `a, b, c` is better cuz the problem is more for beginners, but i can change it if you want

File: solutions/cf-1560B.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:55]: > statements on one line
should i separate the lines?
> also if you don't want overflow, use long long- long is still 32 bits
👍

**Comment by ryanchou-dev** [comment]
> i've seen worse

🥳

File: solutions/cf-1560B.mdx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <stdlib.h>
+using namespace std;
+
+int solve(long a, long b, long c) {
+	long numGroup = abs(a - b) * 2;
```
Comment (maggie-j-liu) [inline solutions/cf-1560B.mdx:34]: `long` in c++ is not necessarily larger than `int`, you probably want `long long`

Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:34]: i thought i changed this

Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:34]: oops

File: solutions/cf-1560B.mdx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <stdlib.h>
+using namespace std;
+
+int solve(long a, long b, long c) {
```
Comment (maggie-j-liu) [inline solutions/cf-1560B.mdx:33]: forgot to change here

**Review by maggie-j-liu — APPROVED** [review]
why is the slug not the same as the id?!?!? i can never find the solution on the deployment

File: solutions/cf-1560B.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/cf-1560B.mdx]: yeah sep them

File: solutions/cf-1560B.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/cf-1560B.mdx:30]: thought abs was alr defined or smth?

Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:30]: oops yeah you're right

File: solutions/cf-1560B.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/cf-1560B.mdx:128]: yeah you're right

File: solutions/cf-1560B.mdx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+			if (ans > numGroup) ans -= (Math.abs(a - b) * 2);
+			return ans;
+		}
+	}
+	
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		for (int i = 0; i < n; i++) {
+			long long a = io.nextInt();
```
Comment (nathangong) [inline solutions/cf-1560B.mdx:93]: wait why are these `long long`?

Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:93]: https://github.com/cpinitiative/usaco-guide/pull/2120#discussion_r692618258

File: solutions/cf-1560B.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+			long long b = io.nextInt();
+			long long c = io.nextInt();
```
Comment (nathangong) [inline solutions/cf-1560B.mdx:95]:
```suggestion
			long a = io.nextInt();
			long b = io.nextInt();
			long c = io.nextInt();
```

File: solutions/cf-1560B.mdx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+</CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class WhosOpposite {
+	static long solve(long long a, long long b, long long c) {
```
Comment (nathangong) [inline solutions/cf-1560B.mdx:71]:
```suggestion
	static long solve(long a, long b, long c) {
```

File: solutions/cf-1560B.mdx [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+		long long numGroup = Math.abs(a - b) * 2;
```
Comment (nathangong) [inline solutions/cf-1560B.mdx:72]:
```suggestion
		long numGroup = Math.abs(a - b) * 2;
```

File: solutions/cf-1560B.mdx [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+	
+		if (a > numGroup || b > numGroup || c > numGroup) {
+			return -1;
+		} else {
+			long long ans = c + Math.abs(a - b);
```
Comment (nathangong) [inline solutions/cf-1560B.mdx:77]:
```suggestion
			long ans = c + Math.abs(a - b);
```

File: solutions/cf-1560B.mdx (hunk 7 above)
Comment (ryanchou-dev) [reply solutions/cf-1560B.mdx:93]: oh this is java

Comment (nathangong) [reply solutions/cf-1560B.mdx:93]: yeah idt long long works or does anything in java

**Comment by ryanchou-dev** [comment]
> why is the slug not the same as the id?!?!? i can never find the solution on the deployment

👀

**Review by nathangong — APPROVED** [review]
java looks good 😉

**Comment by ryanchou-dev** [comment]
> > why is the slug not the same as the id?!?!? i can never find the solution on the deployment
> 
> 👀


where's the slug??

**Comment by maggie-j-liu** [comment]
in the url

**Comment by ryanchou-dev** [comment]
> in the url

this is probably a really dumb question D: (aren't they the same)
image

**Comment by maggie-j-liu** [comment]
i meant the slug on usaco.guide (for this problem, it's probably smth like /problems/**cf-whos-opposite**/solution) which is not the same as the id

**Comment by ryanchou-dev** [comment]
> i meant the slug on usaco.guide (for this problem, it's probably smth like /problems/**cf-whos-opposite**/solution) which is not the same as the id

ohhhh

**Comment by ryanchou-dev** [comment]
> i meant the slug on usaco.guide (for this problem, it's probably smth like /problems/**cf-whos-opposite**/solution) which is not the same as the id

this happens for all the cf probs, right?

**Comment by maggie-j-liu** [comment]
yeah (i think the slug is meant to make the url look better, with problem name instead of an id but it's kind of inconvenient to find the page lol)

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**